### ISSUE TITLE
Line ctx->end = iter->prev may lead to ctx->end being 0 while ctx->begin is not 0.

### ISSUE THREAD

**Issue by Predelnik** [vurtun/nuklear#634]
Here's a link to the line:
https://github.com/vurtun/nuklear/blob/16266100401f8025503371cfa93187ddc299902a/nuklear.h#L18791
I might be able to reproduce minimal example where it leads to segfault but it should be obvious why it's not a good state. And it's also obvious why this state can happen - we have only one window and it's hidden or closed.

As for the fixes I'm not sure. Both ways - not changing `ctx->end` when `iter->prev==0`  or changing `ctx->begin` also to 0 if `ctx->end` became 0 seem to prevent crash for me.

**Comment by vurtun** [comment]
Hey @Predelnik ,
could you create a PR for me and I will pull it.

**Comment by Predelnik** [comment]
@vurtun 
Thanks for the response!
I will make PR, I just wasn't sure which one of the fixes is better.

As for similarity with #637 I wasn't sure as I found them separately, I didn't bother to try to make reproducible example for this one sadly because it clearly seemed to me as a mistake in the code.

Now I checked that applying fix for this issue doesn't change behavior of my example for #637 so it looks like they are unrelated.
